### PR TITLE
Update Platform View docs

### PR DESCRIPTION
**Pull request by blasten**
Adds examples for virtual display.
I will add the iOS examples in a bit, please go ahead and let me know if this looks OK so far.

### CODE REVIEW

File: src/docs/development/platform-integration/platform-views.md [hunk 1]
```diff
@@ -4,30 +4,38 @@ short-title: Platform-views
...
-In 1.22, Hybrid composition is enabled by default,
-and only requires API level 19 (instead of API level 20,
-which the previous version required).
-Also, these updates fix all known keyboard,
-and accessibility issues for Android views.
+Which one to use depends on the use case. Let's take a look:
+
+* Virtual displays renders the native view to a texture,
+  so it's not embeded within the Android view hierachy. Performance 
+  is good although certain platform interactions such as keyboard
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:28]:
```suggestion
Certain platform interactions such as keyboard 
```

File: src/docs/development/platform-integration/platform-views.md [hunk 2]
```diff
@@ -4,30 +4,38 @@ short-title: Platform-views
...
-and accessibility issues for Android views.
+Which one to use depends on the use case. Let's take a look:
+
+* Virtual displays renders the native view to a texture,
+  so it's not embeded within the Android view hierachy. Performance 
+  is good although certain platform interactions such as keyboard 
+  handling, and accessibility features might not work.
+
+* Hybrid composition requires Flutter 1.22. This mode appends the 
+  native view to the Android View hirarchy. Thefore, keyboard
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:32]:
```suggestion
  native view to the Android View hierarchy. Therefore, keyboard
```

File: src/docs/development/platform-integration/platform-views.md [hunk 3]
```diff
@@ -4,30 +4,38 @@ short-title: Platform-views
...
+  handling, and accesibility work out of the box. Performance is
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:33]:
```suggestion
  handling, and accessibility work out of the box.
```

File: src/docs/development/platform-integration/platform-views.md [hunk 4]
```diff
@@ -4,30 +4,38 @@ short-title: Platform-views
...
+  good in Android 10 or above. Prior to Android 10, this mode
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:34]:
```suggestion
  Prior to Android 10, this mode
```

File: src/docs/development/platform-integration/platform-views.md [hunk 5]
```diff
@@ -4,30 +4,38 @@ short-title: Platform-views
...
+  reduces the frame throughput (FPS) of the Flutter UI.
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:35]:
```suggestion
  may significantly reduce the frame throughput (FPS) of the Flutter UI.
```

File: src/docs/development/platform-integration/platform-views.md [hunk 6]
```diff
@@ -212,27 +260,28 @@ For more information, see the API docs for:
...
 ```
 
 ### Performance
 
 Hybrid composition on Android works best on Android 10 or above.
-On lower versions of Android, the Flutter UI can appear slower
-while the native view is rendered on the screen.
-This overhead is due to the synchronization of Flutter frames to the
-Android view system.
+On lower versions of Android, the engine makes a device-host-device
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:278]: Can we spell this out in more detail? I'm not sure what device-host-device copy means?

Is it correct to say something like:
On Android versions older than 10, the hybrid composition mode copies each Flutter frame out of the graphic memory into main memory and then copied back to a GPU texture. As this copy happens per frame the performance of the entire Flutter UI may be impacted significantly.

File: src/docs/development/platform-integration/platform-views.md [hunk 7]
```diff
@@ -212,27 +260,28 @@ For more information, see the API docs for:
...
-        minSdkVersion 19
+        minSdkVersion 19 // if using Hybrid composition.
+        minSdkVersion 20 // if using Virtual display.
     }
 }
 ```
 
 ### Performance
 
 Hybrid composition on Android works best on Android 10 or above.
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:277]: I'll add a section about VD performance, maybe something like:

Each pixel is flowing through additional intermediate graphic buffers (that are as big as the platform view) which cost in graphic memory and drawing perfomance.

Comment (amirh) [reply src/docs/development/platform-integration/platform-views.md:277]: Can we say something about the performance post Android 10? I'm assuming we have similar performance hit to VDs?

**Review by sfshaza2 — COMMENTED** [review]
From my point of view, this PR looks good, though I see @amirh has some requested changes. Let me know when it's ready to merge.

**Comment by blasten** [comment]
@amirh this is ready for another look

**Review by sfshaza2 — APPROVED** [review]
Thanks for your hard work! LGTM

File: src/docs/development/platform-integration/platform-views.md [hunk 8]
```diff
@@ -4,30 +4,37 @@ short-title: Platform-views
...
-and accessibility issues for Android views.
+Which one to use depends on the use case. Let's take a look:
+
+* Virtual displays renders the `android.view.View` instance to a texture,
+  so it's not embedded within the Android Activity's view hierachy.
+  Certain platform interactions such as keyboard handling, and accessibility
+  features might not work.
+
+* Hybrid composition requires Flutter 1.22. This mode appends the
+  native `android.view.View` to the view hirarchy. Therefore, keyboard
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:32]: nit: typo (hierarchy)

File: src/docs/development/platform-integration/platform-views.md [hunk 9]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
-
+[`FlutterPlugin`][]: {{site.api}}/javadoc/io/flutter/embedding/engine/plugins/FlutterPlugin.html
 [`PlatformViewFactory`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformViewFactory.html
 [`PlatformViewRegistry`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html
 [`PlatformView`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformView.html
 
-Finally, modify your `build.gradle` file to require
-version 19 (or later) of the Android API.
+Finally, modify your `build.gradle` file to require either
+the minimal Android SDK version:
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:285]: nit: s/either the/one of the/

File: src/docs/development/platform-integration/platform-views.md [hunk 10]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
-Android view system.
+iOS only uses Hybrid composition, which means that the native
+`UIView` is appended to view hierarchy.
 
-To mitigate this issue, we recommend that you avoid displaying
-the native view while an animation is taking place in Dart.
-For example, you could use a placeholder such as capturing a
-screenshot of the native view and rendering the placeholder
-while the animation is happening.
+In Flutter 1.22, platform views are enabled by default. This means
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:301]: nit: I'd suggest rephrasing this as "Prior to Flutter 1.22 platform views were in developers preview and setting the... flag was required in order to use them on iOS"

File: src/docs/development/platform-integration/platform-views.md [hunk 11]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
+
+### Performance
+
+Platform views in Flutter come with performance trade-offs.
+
+For example, in a typical Flutter app, the Flutter UI is composed
+on a dedicated raster thread. This allows Flutter apps to be fast,
+as the main platform thread is rarely blocked.
+
+While a platform view is rendered, the Flutter UI is composed from
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:511]: Is this always the case or only with hybrid composition?

File: src/docs/development/platform-integration/platform-views.md [hunk 12]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
+the platform thread, which competes with other tasks like
+handling OS or plugin messages, etc.
+
+Prior to Android 10, Hybrid composition copies each Flutter frame
+out of the graphic memory into main memory and then copied back to
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:516]: nit: s/copied back/copies back/

File: src/docs/development/platform-integration/platform-views.md [hunk 13]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
+a GPU texture. As this copy happens per frame, the performance of
+the entire Flutter UI may be impacted.
+
+On the other hand, Virtual display makes each pixel of the native view
+flow through additional intermediate graphic buffers, which cost graphic
+memory and drawing perfomance.
+
+For complex cases, there are some techniques that can be used to mitigate
+these issues.
+
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:526]: Missing section on the performance impact of hybrid composition on Android >= 10

File: src/docs/development/platform-integration/platform-views.md [hunk 14]
```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
...
+handling OS or plugin messages, etc.
+
+Prior to Android 10, Hybrid composition copies each Flutter frame
+out of the graphic memory into main memory and then copied back to
+a GPU texture. As this copy happens per frame, the performance of
+the entire Flutter UI may be impacted.
+
+On the other hand, Virtual display makes each pixel of the native view
+flow through additional intermediate graphic buffers, which cost graphic
+memory and drawing perfomance.
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:522]: nit: typo (performance)

File: src/docs/development/platform-integration/platform-views.md [hunk 15]
```diff
@@ -4,30 +4,37 @@ short-title: Platform-views
 description: Learn how to host native Android and iOS views in your Flutter app with Platform Views.
 ---
 
-As of the Flutter 1.22 release, support for hosting
-Android and iOS views from a Flutter mobile app
-is now production ready.
+Platform views allow to embed native views in a Flutter app.
+
+The Flutter engine takes care of coordinating the platform
+and the Flutter compositors, so you can apply transforms,
```
[the remark this answers is not in the document]
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md:10]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 8 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 9 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 10 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 11 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 12 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 13 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md:527]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 14 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md]: Done.

File: src/docs/development/platform-integration/platform-views.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (amirh) [reply src/docs/development/platform-integration/platform-views.md:10]: ?

File: src/docs/development/platform-integration/platform-views.md [hunk 16]
```diff
@@ -196,58 +243,300 @@ public class MainActivity extends FlutterActivity {
...
+
+### Performance
+
+Platform views in Flutter come with performance trade-offs.
+
+For example, in a typical Flutter app, the Flutter UI is composed
+on a dedicated raster thread. This allows Flutter apps to be fast,
+as the main platform thread is rarely blocked.
+
+While a platform view is rendered with Hybrid composition, the Flutter
```
Comment (amirh) [inline src/docs/development/platform-integration/platform-views.md:511]: nit: it sounds like this is happening while the platform view is visible, which is what we want to ideally have but not where we're currently at

**Review by amirh — APPROVED** [review]
LGTM modulo the last 2 nits

File: src/docs/development/platform-integration/platform-views.md (hunk 16 above)
Comment (blasten) [reply src/docs/development/platform-integration/platform-views.md:511]: This is correct. We merge and unmerge threads.

**Comment by blasten** [comment]
@amirh I'm not sure what you meant by `'?'`. what part is confusing?

File: src/docs/development/platform-integration/platform-views.md (hunk 16 above)
Comment (amirh) [reply src/docs/development/platform-integration/platform-views.md:511]: 👍 
Sorry got confused

**Review by amirh — APPROVED** [review]
LGTM

**Pull request merged**